server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/SetMessagesMethodTest.java

```diff
@@ -1408,4 +1410,46 @@ private boolean isHtmlMessageReceived(AccessToken recipientToken) {
             return false;
         }
     }
+    
+    @Test
+    public void movingAMessageIsNotSupported() throws Exception {
+        jmapServer.serverProbe().createMailbox("#private", username, "heartFolder");
+        Mailbox heartFolder = jmapServer.serverProbe().getMailbox("#private", username, "heartFolder");
+        String heartFolderId = heartFolder.getMailboxId().serialize();
+
+        ZonedDateTime dateTime = ZonedDateTime.parse("2014-10-30T14:12:00Z");
+        jmapServer.serverProbe().appendMessage(username, new MailboxPath("#private", username, "inbox"),
+                new ByteArrayInputStream("Subject: my test subject\r\n\r\ntestmail".getBytes()), Date.from(dateTime.toInstant()), false, new Flags());
+
+        String messageToMoveId = "user|inbox|1";
+
+        String requestBody = "[" +
+                "  [" +
+                "    \"setMessages\","+
+                "    {" +
+                "      \"update\": { \"" + messageToMoveId + "\" : {" +
+                "        \"mailboxIds\": [\"" + heartFolderId + "\"]" +
+                "      }}" +
+                "    }," +
+                "    \"#0\"" +
+                "  ]" +
+                "]";
+
+        given()
+                .accept(ContentType.JSON)
+                .contentType(ContentType.JSON)
+                .header("Authorization", this.accessToken.serialize())
+                .body(requestBody)
+        .when()
+                .post("/jmap")
+        .then()
+                .statusCode(200)
+                .body(NAME, equalTo("messagesSet"))
+                .body(ARGUMENTS + ".notUpdated", hasKey(messageToMoveId))
+                .body(ARGUMENTS + ".notUpdated[\""+messageToMoveId+"\"].type", equalTo("invalidProperties"))
```

Comment:
you should define a variable for `ARGUMENTS + ".notUpdated[\""+messageToMoveId+"\"]`